### PR TITLE
Adding download code for ARCT, MCTest, MCTACO, MuTual, and QuAIL

### PR DESCRIPTION
**Pull request by claravania**
This PR adds download code for the following tasks: arct, mctest160, mctest600, mctaco, mutual, mutual_plus, and quail. Besides quail which uses HF datasets, all tasks use direct download links.

Tested using the following command:
```
export PYTHONPATH=jiant/
python jiant/jiant/scripts/download_data/runscript.py download --tasks [TASK_NAME] --output_path [OUTPUT_PATH]
```

### CODE REVIEW

File: jiant/scripts/download_data/dl_datasets/files_tasks.py [hunk 1]
```diff
@@ -170,6 +194,219 @@ def download_abductive_nli_data_and_write_config(
...
+            os.path.join(task_data_path, f"mc160.{phase}.ans"),
+        )
+    shutil.rmtree(os.path.join(task_data_path, "MCTest"))
+
+    py_io.write_json(
+        data={
+            "task": task_name,
+            "paths": {
+                "train": os.path.join(task_data_path, "mc160.train.tsv"),
+                "train_ans": os.path.join(task_data_path, "mc160.train.ans"),
```
Comment (jeswan) [inline jiant/scripts/download_data/dl_datasets/files_tasks.py:283]: Is the following suitable for the task? I want to try and reduce the number of named keys in a task config (`*_ans` is a new key).
```suggestion
                "train_labels": os.path.join(task_data_path, "mc160.train.ans"),
```

File: jiant/tasks/lib/quail.py [hunk 2]
```diff
@@ -54,8 +54,8 @@ def _create_examples(cls, lines, set_type):
                 Example(
                     guid="%s-%s" % (set_type, i),
                     prompt=line["context"] + " " + line["question"],
-                    choice_list=[d for d in line["answers"]],
-                    label=line["label"],
+                    choice_list=["answers"],
```
Comment (jeswan) [inline jiant/tasks/lib/quail.py:57]: Should this be `line["answers"]`?

Comment (jeswan) [inline jiant/tasks/lib/quail.py:57]:
```suggestion
                    choice_list=[d for d in line["answers"]]
```

**Review by jeswan — CHANGES_REQUESTED** [review]
One concern about QuAIL (see comment). Will approve after this concern and formatting is addressed.

**Comment by claravania** [comment]
@jeswan I've fixed the QuAIL and reformat issues.

**Review by jeswan — APPROVED** [review]

**Pull request merged**